Awesome books: with ES6

Yazino12 opened the pull request:
### In this pull:

- Use modules folder to divide code into junks
- Use export and import
- Refactor functions and methods to arrow functions
- Modify Date function to use Luxon library

**Latest changes:** 

- Uninstall the luxon package and use luxon.js as a file
- Add .eslintignore to ignore code from luxon.js
- Modify date.js to use luxon.js

**Here is the link to version 1 of this project** [See Here](https://github.com/Yazino12/awesome-books)

Yazino12 committed:
Modify awesome books to use es6
Use modules folder to divide code into junks
Use export and import
Refactor functions and methods to arrow functions
Modify Date function to use Luxon library

medaminedev66 commented on modules/book.js:
```diff
@@ -0,0 +1,20 @@
+const book = (bookArray) => {
+  const container = document.querySelector('.container');
+  if (bookArray.length > 0) {
+    bookArray.forEach((book) => {
+      const content = `<div class="book">
+                    <div class="left">
+                      <p>${book.title}</p>
+                      <p>by</p>
+                      <p>${book.author}</p>
+                    </div>
+                      <button class="remove">Remove</button>
+                  </div>`;
+      container.innerHTML += content;
+    });
+  } else {
+    container.innerHTML = '<h3 class="add-books">Add Books!</h3>';
+  }
```
- [x] The navbar links are not working. I'm not able to add books. Kindly fix this issue.

medaminedev66 commented on modules/date.js:
```diff
@@ -0,0 +1,21 @@
+import { DateTime } from '../node_modules/luxon/src/luxon.js';
+
+const displayDate = () => {
+  const dateElement = document.querySelector('.date');
+  const dt = DateTime.now();
+
+  const options = {
+    month: 'long',
+    day: 'numeric',
+    year: 'numeric',
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+  };
+
+  const all = dt.toLocaleString(options);
+
+  dateElement.textContent = `${all}`;
+};
+
+export default displayDate;
```
- [x] The date is not displayed below the navbar. Kindly work on that.
<img width="946" alt="image" src="https://user-images.githubusercontent.com/77190203/159340091-6b1e35e7-be21-4db3-8110-e82868858e97.png">

medaminedev66 reviewed (changes requested):
Hi @Yazino12 , :wave:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️

_See the inline comments below for more required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Yazino12 committed:
Add luxon.js as a file
Uninstall luxon package and use luxon.js
Add .eslintignore to ignore code from luxon.js
Modify date.js to use luxon.js

keroloussamy commented on index.js:
```diff
@@ -184,7 +61,7 @@ class BookCollection {
 }
 
 const arrayCollection = new BookCollection(
-  JSON.parse(localStorage.getItem('bookData'))
+  JSON.parse(localStorage.getItem('bookData')),
 );
```
- [ ] Kindly you still have a problem, because you are depending on getting the data from local storage, but on the reviewer device this data does not exist, that's why we are having the issue and you don't.

The error
![image](https://user-images.githubusercontent.com/35535130/159445192-28a1c72a-41ea-458a-b4cd-4d0fd20cb628.png)


To solve this issue 
replace this line `JSON.parse(localStorage.getItem('bookData')),` with this `JSON.parse(localStorage.getItem('bookData')) || [],`, here you are saying that if there's no data on local storage send empty array.

Yazino12 replied:
@keroloussamy Thanks for pointing out the issue and that walkthrough will definitely help a lot. 🙏🏾✨

keroloussamy reviewed (changes requested):
Hi @Yazino12, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

 ## Required Changes ♻️
_Check the comments under the review._

![](https://camo.githubusercontent.com/6ac905aa3a075ef45de55758c20c6e6efb1dfdbbb0d9fd03d4718d26c5c96cde/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f4a713832345239334a734c775a436169534c2f67697068792e676966)

## Optional suggestions
_Check the comments under the review._

- [x] x

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Yazino12 committed:
Resolve issue with showing data

Yazino12 committed:
Fix eslint error

micheaol reviewed (approved):
# STATUS: APPROVED :green_circle: :clap: 

Hi @Yazino12 , 

Great work on making the changes requested by the previous reviewer 👏 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 
- You set up linter files and no errors. 💯 

![well-1](https://user-images.githubusercontent.com/64173189/128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
